src/js/strategies/DCVWebRTCStrategy.js

```diff
@@ -0,0 +1,148 @@
+        return this.proxy.getUserMedia(constraints);
+    }
+
+    _createPeerConnection(configuration, optionalConfiguration) {
+        return this.proxy.createPeerConnection(configuration, optionalConfiguration);
+    }
+
+    addStream(_pc, stream) {
+        stream.getTracks().forEach(track => {
+            let transceiver = _pc.addTransceiver(track.kind, {
+                streams: [stream]
+            });
+            transceiver.sender.replaceTrack(track);
+        });
+    }
+
+    setRemoteDescription(self, rtcSession) {
+        var setRemoteDescriptionPromise = rtcSession._pc.setRemoteDescription(self._createSessionDescription({
+            type: ANSWER,
+            sdp: self._sdp
+        }));
+        setRemoteDescriptionPromise.catch(e => {
+            self.logger.error('SetRemoteDescription failed', e);
+        });
+        setRemoteDescriptionPromise.then(() => {
+            var remoteCandidatePromises = Promise.all(self._candidates.map(function (candidate) {
+                var remoteCandidate = self._createRemoteCandidate(candidate);
+                self.logger.info('Adding remote candidate', remoteCandidate);
+                return rtcSession._pc.addIceCandidate(remoteCandidate);
+            }));
+            remoteCandidatePromises.catch(reason => {
+                self.logger.warn('Error adding remote candidate', reason);
+            });
+            return remoteCandidatePromises;
+        }).then(() => {
+            rtcSession._sessionReport.setRemoteDescriptionFailure = false;
+            self._remoteDescriptionSet = true;
+            self._checkAndTransit();
+        }).catch(() => {
+            rtcSession._stopSession();
+            rtcSession._sessionReport.setRemoteDescriptionFailure = true;
+            self.transit(new FailedState(rtcSession, RTC_ERRORS.SET_REMOTE_DESCRIPTION_FAILURE));
+        });
+    }
+
+    onIceStateChange(evt, _pc) {
+        return _pc.iceConnectionState;
+    }
+
+    onPeerConnectionStateChange(_pc) {
+        return _pc.connectionState;
+    }
+
+    connect(self) {
+        self._pc.ontrack = hitch(self, self._ontrack);
+    }
+
+    _ontrack(self, evt) {
+        if (evt.streams.length > 1) {
+            console.warn('Found more than 1 streams for ' + evt.track.kind + ' track ' + evt.track.id + ' : ' +
+                evt.streams.map(stream => stream.id).join(','));
+        }
+        let stream = evt.streams[0];
+        self._remoteMediaElement = this.createMediaElement(stream);
+        self._remoteAudioStream = stream;
+        self._remoteMediaElement.srcObject = stream;
+        self._remoteAudioElement.play();
+    }
+
+    createMediaElement(stream) {
+        let props = {
+            kind: AUDIO,
+            autoplay: true,
+        };
+        let element = stream.createMediaElement(props);
+        console.log("Creating proxied media element.");
+        return element;
+    }
+
+    getStrategyName() {
+        return DCV_EXPLICIT;
```

Comment:
is it possible to make the name more consistent with the other two strategies?

Reply:
Updated. Thanks!